### PR TITLE
Remap function names

### PR DESCRIPTION
**Pull request by LindsayWray**
Partially addresses issue #5 

Due to occasional differing function names between DuckDB and Substrait, remapping is necessary. This pr adds unordered maps to remap these function names both ways. These maps will likely need to be expanded in the future.

### CODE REVIEW

File: substrait/from_substrait.cpp [hunk 1]
```diff
@@ -21,6 +21,27 @@
 #include "google/protobuf/util/json_util.h"
 
 namespace duckdb {
+const std::unordered_map<std::string, std::string> SubstraitToDuckDB::function_names_remap = {
```
Comment (pdet) [inline substrait/from_substrait.cpp:24]: Should we have a constexpr for these maps?

File: substrait/from_substrait.cpp [hunk 2]
```diff
@@ -21,6 +21,27 @@
...
+  {"modulus", "mod"},
+  {"std_dev", "stddev"},
+  {"starts_with", "prefix"},
+  {"ends_with", "suffix"}, 
+  {"substring", "substr"},
+  {"char_length", "length"},
+  {"is_nan", "isnan"},
+  {"is_finite", "isfinite"},
+  {"is_infinite", "isinf"},
+  {"like", "~~"}
```
Comment (pdet) [inline substrait/from_substrait.cpp:34]: Is `like` `~~` being tested?

**Comment by pdet** [comment]
Hey Lindsay, many thanks for the PR! It looks great, I only have some small comments.

It also seems you also have to update the tests that match the binary results, since the binaries have now changed :-)

**Comment by pdet** [comment]
Also, is this PR the one that refers to the issue raised on the discord channel?

File: substrait/from_substrait.cpp (hunk 2 above)
Comment (LindsayWray) [reply substrait/from_substrait.cpp:34]: I just added an extra test for like!

File: substrait/from_substrait.cpp (hunk 1 above)
Comment (LindsayWray) [reply substrait/from_substrait.cpp:24]: I looked into it, but unfortunately there is no constexpr constructor support for std::map

**Comment by LindsayWray** [comment]
> Also, is this PR the one that refers to the issue raised on the discord channel?

Yes, I hope this PR resolves that issue

**Comment by pdet** [comment]
The CI is hopefully fixed now :-)

**Comment by pdet** [comment]
LGTM!

**Pull request merged**